dev/source/docs/style-guide.rst

```diff
@@ -558,3 +579,23 @@ For efficiency reasons, ArduPilot doesn't use the C standard library (``std::``)
 This means no ``std::vector`` and no ``std::string``, for example.
 
 If you really want to go that way, including the header to get these isn't sufficient - you will need to fiddle with the build system to link ``std`` in.
+
+Alternatives to ``std::vector``
```

Comment:
maybe we should say that we try to avoid dynamic allocation like on vector and string

Reply:
Added some more words.